modules/transactions/src/main/java/org/apache/ignite/internal/tx/LockException.java

```diff
@@ -17,27 +17,30 @@
 
 package org.apache.ignite.internal.tx;
 
-import org.apache.ignite.lang.IgniteInternalCheckedException;
+import static org.apache.ignite.lang.ErrorGroups.Transactions.RELEASE_LOCK_ERR;
+
+import java.util.UUID;
 
 /**
- * This exception is thrown when a lock cannot be acquired due to conflict.
+ * This exception is thrown when a lock cannot be acquired due to conflict or cannot be released.
  */
-public class LockException extends IgniteInternalCheckedException {
+public class LockException extends TransactionInternalCheckedException {
     /**
-     * The constructor.
+     * Creates a new instance of LockException with the given message.
      *
-     * @param msg The message.
+     * @param msg The detail message.
      */
     public LockException(String msg) {
-        super(msg);
+        super(RELEASE_LOCK_ERR, msg);
     }
 
     /**
-     * Constructor.
+     * Creates a new instance of LockException with the given {@code waiter}, which represents a conflicting transaction.
      *
+     * @param txId Transaction id that trying to acquire or release a lock.
      * @param waiter Conflicting waiter.
      */
-    public LockException(Waiter waiter) {
-        super("Failed to acquire a lock due to a conflict with: " + waiter);
+    public LockException(UUID txId, Waiter waiter) {
+        super(RELEASE_LOCK_ERR, "Failed to acquire a lock due to a conflict [txId=" + txId + ", waiter=" + waiter + ']');
```

Comment:
It's a bit confusing.

- Exception itself seems to be general - it can occur both on accruing and releasing or even within downgrade.
- Error code is about "RELEASE_" only
- Msg is about acquiring: Failed to acquire

Reply:
You are absolutely right. I missed it for some reason unfortunately.